From 9af848d2d76cf34b6f9e7955603c35e058283c47 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 9 Mar 2021 12:36:40 -0800
Subject: [PATCH 1768/1851] drm: xlnx: zynqmp_disp: Enable DP encoder during
 bridge enable

commit f6a1990771100885d014732c3f4dd102e335b344 from
https://github.com/Xilinx/linux-xlnx.git

Currently the video clock and DP encoder are only set through the PS DP
CRTC. To support external CRTC, set the video clock and DP encoder when
the live bridge is enabled.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 8a293589b488..652335bd8745 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -3039,6 +3039,10 @@ static int zynqmp_disp_bridge_enable(struct xlnx_bridge *bridge)
 	if (ret)
 		return ret;
 
+	/* Enable DP encoder if external CRTC attached */
+	if (disp->dpsub->external_crtc_attached)
+		zynqmp_disp_crtc_atomic_enable(crtc, NULL);
+
 	zynqmp_disp_set_g_alpha(disp, disp->alpha_en);
 	zynqmp_disp_set_alpha(disp, disp->alpha);
 	ret = zynqmp_disp_layer_enable(layer->disp, layer,
-- 
2.31.1

